migration-challenge/migration-guide.md

```diff
@@ -0,0 +1,203 @@
+# ES modules migration guide 
+
+## tl;dr
+
+* [How can I update CJS to ESM?](#how-can-i-update-cjs-to-esm)
+* [How can I use Purescript v0.15 on Node.js?](#how-can-i-use-purescript-v015-on-nodejs)
+* [How can I bundle my library or application?](#how-can-i-bundle-my-library-or-application)
+
+## A little bit of background
+
+In April 2021 the Node.js LTS version 10 reached end-of-life, which was the last version that did not yet support ES modules (ESM). This means, that all Node.js LTS and current versions support ES modules. And since all major browsers support ESM already since a long time, there is no need anymore in the JS ecosystem to support Common JS (CJS) and the community is advocating to [drop CJS support](https://github.com/sindresorhus/meta/discussions/15), which kinda makes sense. The community is following suit and is dropping its support for CJS, some notable examples are [remark](https://github.com/remarkjs/remark/tree/main/packages/remark#install), [`node-fetch`](https://github.com/node-fetch/node-fetch#commonjs) and [`framer-motion`](https://github.com/framer/motion) amongst many others. 
+
+What does that mean for Purescript v0.14? Well, a little bit of bad news, because Purescript v0.14 only supports CJS. Fortunately, there has been a [long-standing PR](https://github.com/purescript/purescript/pull/3791) to support ES modules in Purescript. So time to get this over the finish line and get ESM and
+
+## Welcome to Purescript v0.15 with ESM support!
+
+Purescript v0.15 is here and supports ESM with all of its advantages:
+
+* You can use ESM-only libraries
+* A cleaner and simpler way of writing FFI
+* Helping tools to do their job like [DCE](https://en.wikipedia.org/wiki/Dead_code_elimination) or [Code-Splitting](https://reactjs.org/docs/code-splitting.html#code-splitting)
+
+However, this has a couple of implications that will need you to migrate your code and tooling setup:
+
+* v0.15 drops support of CJS
+
+  With no real [reason](#a-little-bit-of-background) to keep supporting CJS, we have decided to drop CJS support alltogether. The maintenance burden would have been just too high to support CJS as well. This means you will need to update your FFI to ESM. More on this here:
+
+  [How can I update CJS to ESM?](#how-can-i-update-cjs-to-esm)
+
+
+* v0.15 drops support for Node.js versions < 12
+  
+  This is just the logical consequence of Node.js versions < 12 having reached EOL and not supporting ESM. More on this here:
+
+  [How can I use Purescript on Node.js?](#how-can-i-update-cjs-to-esm)
+
+* v0.15 drops `purs bundle` and relies on an external bundlers
+  
+  Yes, you heard right. The Purescript compiler no longer comes with a built-in `bundle` command. Updating `purs bundle` to ESM would have required considerable ammount of work, while at the same time being buggy, doing a poor job on bundle size and taking time from the compiler time to work on more useful things.
```

Comment:
I think it's worth stating here that `purs bundle` was actually already broken in a couple of ways. So, this is not going from a "complete working example" to "relying on external used-industry-wide tools" but rather is going from a "buggy unmaintained generally works" to "relying on external used-industry-wide tools"

Reply:
Ok, I tried to update the text with your input. Can you have a look again? Otherwise feel free to just push your ideas into this branch.

Reply:
Yeah, that looks better. Thanks!